### PR TITLE
List template with id no longer works as domain admin

### PR DESCRIPTION
**Pull request by khos2ow**
## Description

Original ticket: https://issues.apache.org/jira/browse/CLOUDSTACK-9862
As domain admin, listTemplates templatefilter=featured no longer work if id= is specified.

Using cloudmonkey with domain-admin credential:

```
(beta2r1-ninja) > list templates templatefilter=featured filter=name,id
count = 9
template:
+--------------------------------------+---------------------------------+
|                  id                  |               name              |
+--------------------------------------+---------------------------------+
| 513b3a6d-c011-46f0-a4a3-2a954cadb673 |      CoreOS Alpha 1367.5.0      |
| 0c04d876-1f85-45a7-b6f4-504de435bf12 |    Debian 8.5 PV base (64bit)   |
| 285f2203-449a-428f-997a-1ffbebbf1382 |           CoreOS Alpha          |
| 332b6ca8-b3d6-42c7-83e5-60fe87be6576 |          CoreOS Stable          |
| 3b705008-c186-464d-ad59-312d902420af |   Windows Server 2016 std SPLA  |
| 4256aebe-a1c1-4b49-9993-de2bc712d521 |       Ubuntu 16.04.01 HVM       |
| 59e6b00a-b88e-4539-aa3c-75c9c7e9fa6c | Ubuntu 14.04.5 HVM base (64bit) |
| 3ab936eb-d8c2-44d8-a64b-17ad5adf8a51 |          CentOS 6.8 PV          |
| 7de5d423-c91e-49cc-86e8-9d6ed6abd997 |          CentOS 7.2 HVM         |
+--------------------------------------+---------------------------------+
(beta2r1-ninja) > list templates templatefilter=featured id=7de5d423-c91e-49cc-86e8-9d6ed6abd997 filter=name,id
Error 531: Acct[b285d62e-0ec2-4a7c-b773-961595ec6356-Ninja-5664] does not have permission to operate within domain id=c9b4f83d-16eb-11e7-a8b9-367e6fe958a9
cserrorcode = 4365
errorcode = 531
errortext = Acct[b285d62e-0ec2-4a7c-b773-961595ec6356-Ninja-5664] does not have permission to operate within domain id=c9b4f83d-16eb-11e7-a8b9-367e6fe958a9
uuidList:
(beta2r1-ninja) > list templates templatefilter=featured ids=7de5d423-c91e-49cc-86e8-9d6ed6abd997 filter=name,id
count = 1
template:
+--------------------------------------+----------------+
|                  id                  |      name      |
+--------------------------------------+----------------+
| 7de5d423-c91e-49cc-86e8-9d6ed6abd997 | CentOS 7.2 HVM |
+--------------------------------------+----------------+
```

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
Testing
- [ ] I have added tests to cover my changes.
- [ ] All relevant new and existing integration tests have passed.
- [ ] A full integration testsuite with all test that can run on my environment has passed.

@blueorangutan package

### CODE REVIEW

**Comment by khos2ow** [comment]
@DaanHoogland @rhtyd @rafaelweingartner The original ticket in Jira filed against 4.10 so I based this PR against that version and merge it forward. If you think we should not do this please advise.

**Comment by DaanHoogland** [comment]
@khos2ow I am fine with that. I don't know if we ever voted in a policy against that but if so, let's revert that.

**Comment by rafaelweingartner** [comment]
@khos2ow you are the contributor. So, in my opinion you should decide. At most, we can suggest what might be better.

I do not think we need to go that far back (4.10), I do not think we will create extra releases for it. As you probably noticed, a lot of issues (if not all) that went to 4.11.1.0 so far, have not gone to 4.10.

File: server/src/com/cloud/api/query/QueryManagerImpl.java [hunk 1]
```diff
@@ -3147,15 +3147,16 @@ private boolean isPermissible(Long accountDomainId, Long offeringDomainId) {
...
-                Account template_acc = _accountMgr.getAccount(template.getAccountId());
-                DomainVO domain = _domainDao.findById(template_acc.getDomainId());
-                _accountMgr.checkAccess(caller, domain);
-
 
-            }// if template is not public, perform permission check here
-            else if (!template.isPublicTemplate() && caller.getType() != Account.ACCOUNT_TYPE_ADMIN) {
-                _accountMgr.checkAccess(caller, null, false, template);
+            // if template is not public, perform permission check here
+            if (!template.isPublicTemplate()) {
```
Comment (rafaelweingartner) [inline server/src/com/cloud/api/query/QueryManagerImpl.java]: Would you mind extracting this block 3151-3159 to a method, document it, and write unit tests?

**Comment by khos2ow** [comment]
Correct, I wasn't sure how far back we're supposed to _support_ if the bug fixes aren't security related. I'm going to rebase against 4.11

**Comment by khos2ow** [comment]
I have completely missed that this is already fixed in #2144 on 4.11! Closing this PR.

**Pull request closed without merging**